### ISSUE TITLE
[1.18.2 - 0.5.4] Glass see through & fog effect (transparency issue)

### ISSUE THREAD

**Issue by ItoKyoshiro** [Asek3/Rubidium#434]
**Minecraft Version**

1.18.2

**Mod Version**

0.5.4

**Forge Version**

40.1.85

**Describe the Issue**

When using glass for Fog Effects works fine while looking at the same height of the player and below, but when looking above player level (in my case, the ceiling) the fog effects turn into chunks of blocks with different transparencies (Image 1 - looking at the ceiling)
![image 1](https://user-images.githubusercontent.com/72139058/225683706-8d46365a-5c7c-4a91-acaa-898fd5871026.png)

As a comparison, when looking at the side the effect is applied properly (and if I put myself up the whole fog effect seems to show better) but when you look above player level it disipates (Image 2 - effect on the side and on top)
![image 2](https://user-images.githubusercontent.com/72139058/225684152-4fc3925b-9a7b-442c-ba1e-8123665912dc.png)

And it's not related to position or chunk but related to where you're looking (Image 3 - looking down from top of glass)
![image 3](https://user-images.githubusercontent.com/72139058/225685075-ee458846-2577-42f0-b8d4-ef3c8bd62026.png)

Another issue (I think it's related to the transparency properties of Glass) is that when you place Dyed Glass you have an effect of see through some blocks. In this case I got the issue with Block Carpentry blocks (frames) where if I place a piece of Glass in front of them they become "see through" and happened with any type of Glass (Vanilla Dyed & Modded ones) except of regular Glass (Image 4 - see through glass effect in front of a block carpentry frame)
![Image 4](https://user-images.githubusercontent.com/72139058/225686490-3247ec34-7f2f-4098-8ec2-d139b5fc668c.png)

The same seems to happen with the Dyed Glass in item form (dropped on the ground) acting like a mini "x-ray" while normal Glass doesn't have this problem so I tested with other mods of glass and all of them have the same "see through" issue (Image 5 - Glass working properly and White Stained Glass on the bottom right working as see through while in item form)
![image 5](https://user-images.githubusercontent.com/72139058/225687254-46a773ae-f88f-46ba-937f-752ea73c0892.png)

And the same effect happens with Nixie Tubes (Create) which have some transparency in the tubes which I believe works similar like Dyed Glass in this case (Image 6 - see through with Nixie Tubes from Create)
![image 6](https://user-images.githubusercontent.com/72139058/225688570-05eab240-6d01-4423-b81f-acaa14d262c6.png)

**TLDR:** I listed the mods which I have tested that this issue happens but I'm pretty sure it happens with all that are "Dyed Glass" (as a contrast, this issue doesn't happens with Quark's Framed Glass but it does with the dyed version of the same blocks (any type of Quark's Died Framed Glass) so everything that is "just" Glass works perfectly but "dyed glass" present the issue. Also it happens when other blocks use the "transparency" like Create's Nixie Tubes.

**Other Mods / Modpack**

Auxiliary Blocks - auxiliaryblocks-1.18.2-0.4.1
BlockCarpentry - blockcarpentry-1.18-0.3.0
Connected Glass - connectedglass-1.1.6-forge-mc1.18
Quark - Quark-3.2-358
Create - create-1.18.2-0.5.0.i

Vault Hunters 3rd Edition - Update 8

**Comment by Asek3** [comment]
Related to transparency sorting issue which cannot be fixed by us
